sqlmesh/core/model/definition.py

```diff
@@ -2361,7 +2351,59 @@ def _refs_to_sql(values: t.Any) -> exp.Expression:
     return exp.Tuple(expressions=values)
 
 
-def _meta_renderer(
+def render_meta_fields(
+    fields: t.Dict[str, t.Any],
+    module_path: Path,
+    path: Path,
+    jinja_macros: t.Optional[JinjaMacroRegistry] = None,
+    macros: t.Optional[MacroRegistry] = None,
+    dialect: DialectType = None,
+    variables: t.Optional[t.Dict[str, t.Any]] = None,
+    default_catalog: t.Optional[str] = None,
+) -> t.Dict[str, t.Any]:
+    def render_field_value(value: t.Any) -> t.Any:
+        if isinstance(value, exp.Expression) or (
+            isinstance(value, str) and d.SQLMESH_MACRO_PREFIX in value
+        ):
+            try:
+                rendered_expr = render_expression(
+                    expression=exp.maybe_parse(value, dialect=dialect),
+                    module_path=module_path,
+                    macros=macros,
+                    jinja_macros=jinja_macros,
+                    variables=variables,
+                    path=path,
+                    dialect=dialect,
+                    default_catalog=default_catalog,
+                )
+                assert rendered_expr and len(rendered_expr) == 1
+                return rendered_expr[0].sql(dialect=dialect)
+            except Exception:
+                pass
+
+        return value
+
+    for field_name, _ in ModelMeta.all_field_infos().items():
+        field_name = field_name[:-1] if field_name.endswith("_") else field_name
+        if (field_value := fields.get(field_name)) and field_name not in {
+            "audits",
+            "signals",
+            "physical_properties",
+            "virtual_properties",
+            "session_properties",
+        }:
+            if isinstance(field_value, t.Dict):
```

Comment:
Hm, not sure whether it's a good practice to check for instance of `t.Dict`. I believe we just check for `dict` in other places.